qpc/report/insights.py

```diff
@@ -122,7 +127,11 @@ def _validate_args(self):
 
     def _handle_response_success(self):
         try:
-            write_file(self.args.path, self.response.content, binary=True)
+            if self.args.path:
+                file_content = self.response.content
+            else:
+                file_content = extract_json_from_tar(self.response.content)
```

Comment:
So... in the end we still need this if/else.
why? setting up request headers didn't work? :thinking:

Reply:
response.content is returning a binary. I have to investigate.

Reply:
maybe we are required to pass the `format=json` queryparam (IoW: `/path/to/endpoint?format=json`)